src/Raven.Server/Documents/Handlers/DocumentReplicationHandler.cs

```diff
@@ -172,20 +171,26 @@ private long GetLastReceivedEtag(Guid srcDbId, DocumentsOperationContext context
             return vectorEntry.Etag;
         }
 
-	    private void WriteReplicationBatchAcknowledge(WebsocketStream webSocketStream, 
+	    private async Task WriteReplicationBatchAcknowledge(WebsocketStream webSocketStream, 
 			DocumentsOperationContext context,
 			bool hasSucceeded)
 	    {
 		    try
 		    {
-			    using (var responseWriter = new BlittableJsonTextWriter(context, webSocketStream))
+			    var responseWriter = new BlittableJsonTextWriter(context, webSocketStream);
+			    try
 			    {
 				    context.Write(responseWriter, new DynamicJsonValue
 				    {
 					    [Constants.MessageType] = Constants.Replication.MessageTypes.ReplicationBatchAcknowledge,
 					    [Constants.HadSuccess] = hasSucceeded
 				    });
 			    }
+			    finally
+			    {
+					//WebSocketStream does not support synchronous writes to the stream
+				    await responseWriter.DisposeAsync();
```

Comment:
Dispose uses Flush which uses stream's sync Write(). Here I want to use WriteSAsync() -> so I implemented DisposeAsync() -> FlushAsync() -> WriteSAsync()